benchmarks/viscoelastic_plastic_shear_bands/gerya_2019/gerya_2019_analysis.py

```diff
@@ -0,0 +1,109 @@
+# Python script to load the solution data into a
+# numpy array, interpolate the data to a new
+# uniform grid, and plot the results.
+# From a terminal, execute the script with
+#   python gerya_2019_analysis.py
+# Within a python interpreter (e.g., a Jupyter notebook), 
+# the script can be executed with the command:
+#   exec(open("gerya_2019_analysis.py").read())
+# This script was tested with the following 
+# package versions:
+#   python     - 3.7.7
+#   numpy      - 1.19.1
+#   scipy      - 1.5.2
+#   vtk        - 8.2.0
+#   matplotlib - 3.3.1
+
+# Load modules
+import numpy as np
+import matplotlib.pyplot as plt
+import vtk as vtk; from vtk.util import numpy_support
+from scipy.interpolate import griddata
+
+# Relative (from current directory) path to the
+# solution data folder.
+vtk_directory = 'output_gerya_2019_vp/solution/'
+
+# Time step to analyze. This number corresponds to 
+# the numbering of .pvtu files. In this instance
+# '00000.0213' is the last nonlinear iteration
+# of the first time step, but note that the 
+# number of nonlinear iterations may vary slightly
+# between machines, or significantly when additional
+# parameters are altered.
+pvtu_number = '00000.0213'
+
+# Load vtu data (pvtu directs to vtu files)
+reader = vtk.vtkXMLPUnstructuredGridReader()
+reader.SetFileName(vtk_directory + 'solution-' + pvtu_number + '.pvtu')
+reader.Update()
+
+# Get the coordinates of nodes in the mesh
+nodes_vtk_array= reader.GetOutput().GetPoints().GetData()
+
+# Convert nodal vtk data to a numpy array
+nodes_numpy_array = vtk.util.numpy_support.vtk_to_numpy(nodes_vtk_array)
+
+# Extract x, y and z coordinates from numpy array 
+x,y = nodes_numpy_array[:,0] , nodes_numpy_array[:,1]
+
+# Determine the number of scalar fields contained in the .pvtu file
+number_of_fields = reader.GetOutput().GetPointData().GetNumberOfArrays()
+
+# Determine the name of each field and place it in an array.
+field_names = []
+for i in range(number_of_fields):
+  field_names.append(reader.GetOutput().GetPointData().GetArrayName(i))
+
+# Determine the index of the field strain_rate
+idx = field_names.index("strain_rate")
+
+# Extract values of strain_rate
+field_vtk_array = reader.GetOutput().GetPointData().GetArray(idx)
+strain_rate     = numpy_support.vtk_to_numpy(field_vtk_array)
+
+# Generate coordinate arrays for data interpolation based on model dimensions
+# and grid spacing. The coordinate arrays will have the double the resolution
+# as the grid spacing.
+res = x[1] - x[0]
+xi = np.linspace(np.min(x), np.max(x), int((np.max(x) - np.min(x))/res)*2 + 1 )
+yi = np.linspace(np.min(y), np.max(y), int((np.max(y) - np.min(y))/res)*2 + 1 )
+
+# Generate data interpolation grid. This is a helpful step as the VTU data format
+# has more than value specified at each grid point. This is due to setting
+# 'set Interpolate output = true' in ASPECT.
```

Comment:
I thought that was due to the general format of how we write the output (we write out unstructured meshes, and so we have to give the locations of the corners of each cell). Bt maybe you don't mean what I think you mean here.

Reply:
You are probably correct that vertices are repeated even without the interpolate output option. Regardless, in the raw VTU data specific points (vertices) are list more than once, and as you stated this is almost certainly due to the way we write out unstructured data. Interpolating the output in ASPECT should then just cause further repetition of some vertices. In any case, as noted above, I removed this comment to avoid confusion :)